src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -310,13 +311,18 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 
 					ApiResponse res;
 					switch (reqType) {
-					case action:	
-						// TODO Handle POST requests - need to read these in and then parse params from POST body
-						/*
-						if (getOptionsParamApi().isPostActions()) {
-							throw new ApiException(ApiException.Type.DISABLED);
+					case action:
+						if (requestHeader.getMethod().equalsIgnoreCase(HttpRequestHeader.POST)) {
+							String contentTypeHeader = requestHeader.getHeader(HttpHeader.CONTENT_TYPE);
+							if(contentTypeHeader != null &&
+									contentTypeHeader.equals(HttpHeader.FORM_URLENCODED_CONTENT_TYPE)) {
+								String body = httpIn.readRequestBody(requestHeader).toString();
+								params = getParams(body);
+							} else {
+								throw new ApiException(ApiException.Type.BAD_TYPE);
```

Comment:
If we're talking exception type then yes I agree, a "new" more specific type would be great.